HyperVExtension/test/HyperVExtension/HyperVExtensionTests/Services/HyperVExtensionIntegrationTest.cs

```diff
@@ -293,4 +291,49 @@ private void PrintProgressData(ConfigurationSetChangeData progressData)
 
         System.Diagnostics.Trace.WriteLine(sb);
     }
+
+    [TestMethod]
+    public void TestDevSetupAgentDeployment()
+    {
+        IHyperVManager hyperVManager = TestHost!.GetService<IHyperVManager>();
+        var machines = hyperVManager.GetAllVirtualMachines();
+        HyperVVirtualMachine? testVm = null;
+        foreach (var vm in machines)
+        {
+            if (string.Equals(vm.Name, "TestVM", StringComparison.OrdinalIgnoreCase))
+            {
+                testVm = vm;
+                break;
+            }
+        }
+
+        Assert.IsNotNull(testVm);
+
+        var powerShell = TestHost!.GetService<IPowerShellService>();
+
+        var deploymentHelperMock = new Mock<DevSetupAgentDeploymentHelper>(powerShell, testVm.Id);
+        deploymentHelperMock.CallBase = true;
+        deploymentHelperMock.Setup(x => x.GetSourcePath(It.IsAny<ushort>())).Returns(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\..\..\..\DevSetupAgent.zip"));
+
+        // TODO: figure out how to get the password from the user
+        var userName = "LocalAdmin";
+        var pwd = new NetworkCredential(string.Empty, "<>").SecurePassword;
+        deploymentHelperMock.Object.DeployDevSetupAgent(userName, pwd);

```

Comment:
I think Eric was saying something about cred scan in the pipelines not liking the use of even test usernames and passwords. So just in case we should probably do it like this, to obfuscate it a little more:

```Suggestion
        deploymentHelperMock.Object.DeployDevSetupAgent("LocalAdmin", new NetworkCredential(string.Empty, "<>").SecurePassword);
```

Reply:
How does it obfuscate it? Or do you know how that detection works? Those values don't matter as you have to change them into real ones. I can just use string.empty.

Reply:
I was thinking more that the cred scan would be looking for variable names like "username" and "pwd" within a function.